06.bowling_object/game.rb

```diff
@@ -5,20 +5,27 @@
 
 class Game
   def initialize(input_text)
-    shots = input_text.split(',').map { |mark| Shot.new(mark) }
-    count = 0
-    @frames = []
-    while @frames.size < 9
-      shots_per_frame = shots[count].strike? ? 1 : 2
-      @frames << Frame.new(@frames.size, *shots[count, shots_per_frame])
-      count += shots_per_frame
-    end
-    @frames << Frame.new(@frames.size, *shots[count..])
+    @input_text = input_text
   end
 
   def calc_total_score
-    @frames.sum do |frame|
-      frame.calc_total_score(@frames)
+    frames = devide_frames
+    frames.sum do |frame|
+      frame.calc_total_score(frames)
+    end
+  end
+
+  private
+
+  def devide_frames
+    shots = @input_text.split(',').map { |mark| Shot.new(mark) }
+    count = 0
+    frames = []
+    while frames.size < 9
+      shots_per_frame = shots[count].strike? ? 1 : 2
+      frames << Frame.new(frames.size, *shots[count, shots_per_frame])
+      count += shots_per_frame
     end
+    frames << Frame.new(frames.size, *shots[count..])
```

Comment:
```suggestion
    [*frames, Frame.new(frames.size, *shots[count..])]
```

好みの問題かもですが、僕はこういう書き方の方が良いかなーと思ってます。
https://qiita.com/jnchito/items/91bda171adceca23abc3

Reply:
ありがとうございます！
戻り値の表現にも留意していきたいと思います！